### PR TITLE
[Mac] Initial ObjectName implementation.

### PR DESCRIPTION
**Pull request by CartBlanche**
Closes - https://github.com/xamarin/Xamarin.PropertyEditing/issues/87

### CODE REVIEW

**Comment by ermau** [comment]
Additionally, there should not be a property button for the name property.

File: Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs [hunk 1]
```diff
@@ -364,6 +359,11 @@ private void UpdateMembers (IObjectEditor[] removedEditors = null, IObjectEditor
 
 			SetNameable (firstNameable);
 			SetCurrentObjectName (name, this.objEditors.Count > 1);
+
+			UpdateProperties (newPropertySet, removedEditors, newEditors);
```
Comment (ermau) [inline Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs]: Where is the test covering these changes?

Comment (CartBlanche) [reply Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs]: In my haste, I totally forgot. I'll add them today.

Comment (CartBlanche) [reply Xamarin.PropertyEditing/ViewModels/PropertiesViewModel.cs]: This change was unnecessary. Reverted, so unit tests will stay as is.

File: Xamarin.PropertyEditing/Reflection/ReflectionPropertyInfo.cs [hunk 2]
```diff
@@ -69,18 +69,24 @@ public virtual async Task SetValueAsync<T> (object target, T value)
...
+					if (typeof (T) == typeof (string))
+						value = value.ToString ();
+					else
+						value = Convert.ChangeType (value, typeof (T));
+				}
+
+				return (T)value;
+
+			} catch (Exception ex) {
+				return default (T);
```
[the remark this answers is not in the document]
Comment (CartBlanche) [reply Xamarin.PropertyEditing/Reflection/ReflectionPropertyInfo.cs]: IIRC it was part of the the other change that I had to remove. This has now been removed.

**Comment by CartBlanche** [comment]
@ermau The Header should now appear during Categorisation. Note the category expansion won't work until the changes you mentioned a week or so ago, are in master.

**Comment by ermau** [comment]
All of them should work except Brush

**Comment by ermau** [comment]
@CartBlanche this looks like a bad merge, all the `.Select()` changes are removed in this PR now.

**Comment by CartBlanche** [comment]
@ermau Sorry which `.Select()` changes?

**Comment by ermau** [comment]
@CartBlanche look at the very first change in the PR diff. See also: #388

**Comment by CartBlanche** [comment]
@ermau that looks correct now. I've also rebased the TypeName and IconView PRs off this one, so they should also be ready once this PR is fine.

File: Xamarin.PropertyEditing.Mac/PropertyTableDelegate.cs [hunk 3]
```diff
@@ -190,46 +203,55 @@ public nint GetHeight (PropertyViewModel vm)
...
 		}
 
 		private void GetVMGroupCellItendifiterFromFacade (NSObject item, out PropertyViewModel vm, out IGroupingList<string, EditorViewModel> group, out string cellIdentifier)
 		{
 			var facade = (NSObjectFacade)item;
 			vm = facade.Target as PropertyViewModel;
 			group = facade.Target as IGroupingList<string, EditorViewModel>;
-			cellIdentifier = (group == null) ? vm.GetType ().FullName : group.Key;
+			cellIdentifier = facade.Target is PanelHeaderEditorControl pvh
+				? "PanelHeaderEditorControl"
```
Comment (ermau) [inline Xamarin.PropertyEditing.Mac/PropertyTableDelegate.cs]: `nameof()` ?

**Comment by CartBlanche** [comment]
@ermau The name over change was pushed.

**Comment by ermau** [comment]
This breaks categories, none but Brush show up.

**Comment by CartBlanche** [comment]
@ermau Really sorry about letting something so stupid slip through. Here's a screencast testing all the various modes - https://www.screencast.com/t/faxRmNdy with the latest changes.

**Pull request merged**